components/fxa-client/android/src/main/java/mozilla/appservices/fxaclient/FirefoxAccount.kt

```diff
@@ -322,12 +322,22 @@ class FirefoxAccount(handle: FxaHandle, persistCallback: PersistCallback?) : Aut
      * This performs network requests, and should not be used on the main thread.
      */
     fun migrateFromSessionToken(sessionToken: String, kSync: String, kXCS: String): JSONObject {
-        val json = rustCallWithLock { e ->
-            LibFxAFFI.INSTANCE.fxa_migrate_from_session_token(this.handle.get(), sessionToken, kSync, kXCS, false, e)
-        }.getAndConsumeRustString()
-
-        this.tryPersistState()
-        return JSONObject(json)
+        try {
+            val json = rustCallWithLock { e ->
+                LibFxAFFI.INSTANCE.fxa_migrate_from_session_token(
+                    this.handle.get(),
+                    sessionToken,
+                    kSync,
+                    kXCS,
+                    false,
+                    e
+                )
+            }.getAndConsumeRustString()
+            return JSONObject(json)
+        } finally {
```

Comment:
Is it a good idea to do this indiscriminately of the exception type? E.g. any way the throwing code may leave around a corrupt or non-complete state, that will cause issues during initialization later?

Reply:
Hrm, I can't think of any, but that doesn't mean one doesn't exist.

Reply:
I've kept the `finally`, but updated the code to make it a bit clearer that this is expected to be safe (and lightly changed up the underlying rust code to make that clearer as well).